[CI] Reusable CI

orensbruli opened the pull request:
Signed-off-by: Esteban Martinena <orensbruli@gmail.com>

Modified build.yaml workflow to use reusable workflow.

luca-della-vedova commented:
`main` is not compatible with galactic / foxy so we should probably have the workflow only do humble

I noticed your commits say "unverified", did you upload your gpg key to github?

orensbruli commented:
I have updated [rmf_ci_templates](https://github.com/open-rmf/rmf_ci_templates/blob/main/.github/workflows/reusable_build.yaml) to accept the workflow matrix. Now we can limit the version in which we want to run a specific workflow. In this case, I have removed the foxy and galactic ones.

luca-della-vedova reviewed (approved):
Great, thanks for cleaning this up!
